### PR TITLE
feat: customize messages displayed to the user (STARGATE-1732)

### PR DESCRIPTION
**Pull request by lukewhyte**

### CODE REVIEW

File: packages/pci-sdk-iframe/src/pages/app/App.test.tsx [hunk 1]
```diff
@@ -142,6 +142,36 @@ describe('<App />', () => {
...
+
+				stubMultipleJSONRespones([
+					{ httpStatus: 400, body: {} },
+					{ httpStatus: 200, body: dummyRequest2FACodeResponse },
+					{ httpStatus: 200, body: getDummyverify2FACodeResponse('failed') }
+				]);
+
+				fireEvent(window, new MessageEvent('message', { data: JSON.stringify({ type: 'showCardData' }) }));
+
+				await waitFor(() => expect(window.alert).toHaveBeenCalledWith('custom_tooManyAttemptsMessage'));
```
Comment (IagoLast) [inline packages/pci-sdk-iframe/src/pages/app/App.test.tsx:173]: we do only care of 2 out 4 error messages? :trollface:

File: packages/pci-sdk-iframe/src/pages/app/useApp.ts [hunk 2]
```diff
@@ -33,6 +37,10 @@ export default function useApp() {
 		labelPan: urlParams.get('labelPan') as string || 'Card number',
 		lastFour: urlParams.get('lastFour') as string || '••••',
 		nameOnCard: urlParams.get('nameOnCard') as string || '',
+		expiredMessage: urlParams.get('expiredMessage') as string || 'Process expired. Start again.',
+		tooManyAttemptsMessage: urlParams.get('tooManyAttemptsMessage') as string || 'Too many attempts, try again.',
+		enter2FAPrompt: urlParams.get('enter2FAPrompt') as string || 'Enter the code we sent you:',
+		failed2FAPrompt: urlParams.get('failed2FAPrompt') as string || 'Wrong code. try again:',
```
Comment (IagoLast) [inline packages/pci-sdk-iframe/src/pages/app/useApp.ts]: 🌈 maybe defaults should be constants

**Review by IagoLast — COMMENTED** [review]
LGTM just one question: are we leaving the public API for a different PR?

**Comment by lukewhyte** [comment]
> LGTM just one question: are we leaving the public API for a different PR?

I can add it to this PR

**Review by IagoLast — CHANGES_REQUESTED** [review]
- Missing test cases
- Lint error

**Comment by IagoLast** [comment]
> > LGTM just one question: are we leaving the public API for a different PR?
> 
> I can add it to this PR

Nah, its ok like this

**Comment by lukewhyte** [comment]
> * Missing test cases
> * Lint error

Lint error is fixed. Adding the missing test cases is a larger conversation. They're used in window.prompt, which is not supported by JSDOM. Thus, we'd have to build our own version of prompt (_[I ran across an old demo of such while investigating this](https://gist.github.com/apeckham/404170)_) and test our error messages in that context.

**Review by IagoLast — COMMENTED** [review]
I was thinking about this and I'm not sure if we want to persist those messages in the state...
I think it complicates the logic and we know it's not going to change (we don't have any methods for that).

Also I've working on a refactor that I want to discuss with you first

**Review by IagoLast — APPROVED** [review]

**Pull request merged**